azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -41,15 +72,176 @@ def _log_policy(cls, msg, is_success=True, op=WALAEventOperation.Policy, send_ev
         else:
             logger.error(msg)
         if send_event:
-            add_event(op=op, message=msg, is_success=is_success)
+            add_event(op=op, message=msg, is_success=is_success, log_event=False)
 
     @staticmethod
-    def is_policy_enforcement_enabled():
+    def __get_policy_enforcement_enabled():
         """
-        Check whether user has opted into policy enforcement feature.
-        Caller function should check this before performing any operations.
+        Policy will be enabled if (1) policy file exists at _CUSTOM_POLICY_PATH and (2) the conf flag "Debug.EnableExtensionPolicy" is true.
         """
-        # TODO: Add check for policy file present at /etc/waagent_policy.json.
         # Policy should only be enabled if conf flag is true AND policy file is present.
-        return conf.get_extension_policy_enabled()
+        policy_file_exists = os.path.exists(_CUSTOM_POLICY_PATH)
+        return conf.get_extension_policy_enabled() and policy_file_exists
+
+    @property
+    def policy_enforcement_enabled(self):
+        return self._policy_enforcement_enabled
+
+    def __get_policy(self):
```

Comment:
Should be static. It is called from init and, as the code evolves, it may end up referencing an uninitialized instance member

Reply:
not sure I understand - what's an example of when this could happen, and would it not be caught by unit tests? I've changed the code in __parse_policy to directly update self._policy, trying to understand if this implementation is problematic.

Reply:
Makes sense, thanks for explaining. I've updated the implementation to make both __get_policy() and __parse_policy() static, since they are both called from init.